src/library/scala/collection/immutable/Vector.scala

```diff
@@ -110,6 +114,45 @@ final class Vector[+A] private[immutable] (private[collection] val startIndex: I
     else
       throw new IndexOutOfBoundsException(index.toString)
   }
+  // requires structure is at pos oldIndex = xor ^ index
+  private final def getElem(index: Int, xor: Int): A = {
+    if (xor < (1 << 5)) { // level = 0
+      (display0
+      (index & 31).asInstanceOf[A])
```

Comment:
This is really confusing indenting.  Try instead
```
display0(
  index & 31
).asInstanceOf[A]
```
and then expand the internal section as needed.  The way you have it here, you have expressions of different levels of nesting at the same indentation.

Reply:
(Note that the original was somewhat sensibly indented, though not according to my recommendation.)

Reply:
I did just C&P in intellij. It would be good to get intellij formatter to align with your expectations

I have adjusted to a format that show the progression - different to the one you suggest as I don't see this allows a progression to the more complex cases

Reply:
the vector changes are also in https://github.com/scala/scala/pull/6987 to aid the review of the Vector focussed changes